src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -110,6 +134,117 @@ public function loadSubtreeAction(Request $request): Root
+
+        return new NodeExtendedInfo($locationPermissionRestrictions);
+    }
+
+    /**
+     * @return TPermissionRestrictions
+     *
+     * @throws \Ibexa\AdminUi\Exception\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    public function getLocationPermissionRestrictions(Location $location): array
+    {
+        $lookupCreateLimitationsResult = $this->permissionChecker->getContentCreateLimitations($location);
+        $lookupUpdateLimitationsResult = $this->permissionChecker->getContentUpdateLimitations($location);
+
+        $createLimitationsValues = $this->lookupLimitationsTransformer->getGroupedLimitationValues(
+            $lookupCreateLimitationsResult,
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+
+        $updateLimitationsValues = $this->lookupLimitationsTransformer->getGroupedLimitationValues(
+            $lookupUpdateLimitationsResult,
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+
+        return [
+            'create' => [
+                'hasAccess' => $lookupCreateLimitationsResult->hasAccess(),
+                'restrictedContentTypeIds' => $createLimitationsValues[Limitation::CONTENTTYPE],
+                'restrictedLanguageCodes' => $createLimitationsValues[Limitation::LANGUAGE],
+            ],
+            'edit' => [
+                'hasAccess' => $lookupUpdateLimitationsResult->hasAccess(),
+                'restrictedLanguageCodes' => $updateLimitationsValues[Limitation::LANGUAGE],
+            ],
+            'delete' => [
+                'hasAccess' => $this->canUserRemoveContent($location),
+            ],
+            'hide' => [
+                'hasAccess' => $this->canUserHideContent($location),
+            ],
+        ];
+    }
+
+    /**
+     * @throws \Ibexa\AdminUi\Exception\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     */
+    private function canUserRemoveContent(Location $location): bool
+    {
+        $content = $location->getContent();
+        $contentType = $content->getContentType();
+        $contentIsUser = (new ContentTypeIsUser($this->configResolver->getParameter('user_content_type_identifier')))
+            ->isSatisfiedBy($contentType);
+
+        $translations = $content->getVersionInfo()->languageCodes;
+        $target = (new Target\Version())->deleteTranslations($translations);
+
+        if ($contentIsUser) {
+            return $this->permissionResolver->canUser(
+                'content',
+                'remove',
+                $content,
+                [$target]
+            );
+        }
+
+        if ($location->depth !== 1) {
```

Comment:
```suggestion
        if ($location->depth > 1) {
```
as it applies to any depth over root.